### PR TITLE
Convert VOLATILE from CREATE arg to property

### PR DESCRIPTION
**Pull request by treysp**
Motivating problem: Teradata `CREATE VOLATILE TABLE` should transpile to `CREATE TABLE` for all other dialects (from #1447).

This PR converts VOLATILE from a CREATE arg to a property and uses the unsupported property machinery to suppress it for other dialects.

### CODE REVIEW

File: sqlglot/generator.py [hunk 1]
```diff
@@ -200,10 +201,11 @@ class Generator:
         exp.SetProperty: exp.Properties.Location.POST_CREATE,
         exp.SortKeyProperty: exp.Properties.Location.POST_SCHEMA,
         exp.SqlSecurityProperty: exp.Properties.Location.POST_CREATE,
+        exp.StabilityProperty: exp.Properties.Location.POST_SCHEMA,
         exp.TableFormatProperty: exp.Properties.Location.POST_WITH,
         exp.TemporaryProperty: exp.Properties.Location.POST_CREATE,
         exp.TransientProperty: exp.Properties.Location.POST_CREATE,
-        exp.VolatilityProperty: exp.Properties.Location.POST_SCHEMA,
+        exp.VolatileProperty: exp.Properties.Location.UNSUPPORTED,
```
Comment (tobymao) [inline sqlglot/generator.py:208]: don't make general code unsupported

File: sqlglot/parser.py [hunk 2]
```diff
@@ -1194,6 +1192,13 @@ def _parse_property_before(self) -> t.Optional[exp.Expression]:
         return None
 
     def _parse_property(self) -> t.Optional[exp.Expression]:
+        if self._prev.token_type in (
```
Comment (tobymao) [inline sqlglot/parser.py:1195]: why is this needed?, why don't you treat it like a normal property

Comment (GeorgeSittas) [reply sqlglot/parser.py:1195]: Should we factor these tokens out into a class constant set?

Comment (tobymao) [reply sqlglot/parser.py:1195]: why don't you just use self._index ?

Comment (tobymao) [reply sqlglot/parser.py:1195]: can you describe to me exactly with sql examples why you need this?

Comment (tobymao) [reply sqlglot/parser.py:1195]: perfect

Comment (tobymao) [reply sqlglot/parser.py:1195]: can you make the change that you proposed and i can review that

**Review by tobymao — APPROVED** [review]
